rs/crypto/src/vetkd/mod.rs

```diff
@@ -0,0 +1,560 @@
+    })
+}
+
+fn share_to_clib_share(
+    share: &ic_types::crypto::vetkd::VetKdEncryptedKeyShare,
+) -> Result<ic_crypto_internal_bls12_381_vetkd::EncryptedKeyShare, VetKdKeyShareCombinationError> {
+    let encrypted_key_share_192_bytes = {
+        if share.encrypted_key_share.0.len() != EncryptedKeyShare::BYTES {
+            return Err(VetKdKeyShareCombinationError::InvalidArgumentEncryptedKeyShare);
+        }
+        let mut encrypted_key = [0u8; EncryptedKeyShare::BYTES];
+        encrypted_key.copy_from_slice(&share.encrypted_key_share.0);
+        encrypted_key
+    };
+    ic_crypto_internal_bls12_381_vetkd::EncryptedKeyShare::deserialize(
+        encrypted_key_share_192_bytes,
+    )
+    .map_err(|e| match e {
+        EncryptedKeyShareDeserializationError::InvalidEncryptedKeyShare => {
+            VetKdKeyShareCombinationError::InvalidArgumentEncryptedKeyShare
+        }
+    })
+}
+
+fn verify_encrypted_key_internal(
+    lockable_threshold_sig_data_store: &LockableThresholdSigDataStore,
+    key: &VetKdEncryptedKey,
+    args: &VetKdArgs,
+) -> Result<(), VetKdKeyVerificationError> {
+    let encrypted_key = {
+        let encrypted_key_192_bytes = {
+            if key.encrypted_key.len() != EncryptedKey::BYTES {
+                return Err(VetKdKeyVerificationError::InvalidArgumentEncryptedKey);
+            }
+            let mut encrypted_key = [0u8; EncryptedKey::BYTES];
+            encrypted_key.copy_from_slice(&key.encrypted_key);
+            encrypted_key
+        };
+        ic_crypto_internal_bls12_381_vetkd::EncryptedKey::deserialize(encrypted_key_192_bytes)
+        .map_err(|e| match e {
+            ic_crypto_internal_bls12_381_vetkd::EncryptedKeyDeserializationError::InvalidEncryptedKey => VetKdKeyVerificationError::InvalidArgumentEncryptedKey,
+        })?
+    };
+
+    let master_public_key = {
+        let pub_coeffs_from_store = lockable_threshold_sig_data_store
+            .read()
+            .transcript_data(&args.ni_dkg_id)
+            .map(|data| data.public_coefficients().clone())
+            .ok_or_else(|| {
+                VetKdKeyVerificationError::ThresholdSigDataNotFound(
+                    ThresholdSigDataNotFoundError::ThresholdSigDataNotFound {
+                        dkg_id: args.ni_dkg_id.clone(),
+                    },
+                )
+            })?;
+        let public_key_bytes = match pub_coeffs_from_store {
+            PublicCoefficients::Bls12_381(pub_coeffs) => pub_coeffs
+                .coefficients
+                .iter()
+                .copied()
+                .next()
+                .ok_or_else(|| {
+                    VetKdKeyVerificationError::InternalError(format!(
+                        "public coefficients for NI-DKG ID {} are empty",
+                        &args.ni_dkg_id
+                    ))
+                })?,
+        };
+        ic_crypto_internal_bls12_381_vetkd::G2Affine::deserialize(&public_key_bytes).map_err(
```

Comment:
Can we use `deserialize_unchecked` here? (Unsure)

Reply:
Good question. Also unsure. The G2 bytes we are deserializing are coming from our threshold sig data store. They were put there by us, but we put there whatever callers of `NiDkgAlgorithm::load_transcript` provide. Likely the only caller of this is Consensus. This might be good enough, but not fully sure yet. We can start with `G2Affine::deserialize` for now and evaluate separately whether we want to use `G2Affine::deserialize_unchecked` instead. I created https://dfinity.atlassian.net/browse/CRP-2688.

There is another instance of that in `combine_encrypted_key_shares_internal` where we deserialize an individual node's G2 public key, where this public key is calculated by us with `lazily_calculated_public_key_from_store`, so there we also could consider using `deserialize_unchecked`.